File: src/types/Opti.ts
import { Point } from "./Point"

export class Opti {
  constructor(
    public penalty: number = 0,
    public c: Point[] = [new Point(), new Point()],
    public t: number = 0,
    public s: number = 0,
    public alpha: number = 0
  ) {
    this.penalty = penalty
    this.c = c
    this.t = t
    this.s = s
    this.alpha = alpha
  }
}

File: src/types/opti.rs
use super::point::Point;
#[derive(Clone)]
pub struct Opti {
    __constructor: (),
    pub pen: usize,
    pub c: Vec<Point>,
    pub t: f64,
    pub s: f64,
    pub alpha: f64,
}

impl Opti {
    pub fn default() -> Opti {
        Opti {
            __constructor: (),
            pen: 0,
            c: vec![Point::default(), Point::default()],
            t: 0.0,
            s: 0.0,
            alpha: 0.0,
        }
    }
}
